jellyfin_kodi/full_sync.py

```diff
@@ -405,38 +412,42 @@ def music(self, library, dialog):
         with self.library.music_database_lock:
             with Database('music') as musicdb:
                 with Database('jellyfin') as jellyfindb:
-                    obj = Music(self.server, jellyfindb, musicdb, self.direct_path)
-
-                    for items in server.get_artists(library['Id'], False, self.sync['RestorePoint'].get('params')):
-
-                        self.sync['RestorePoint'] = items['RestorePoint']
-                        start_index = items['RestorePoint']['params']['StartIndex']
-
-                        for index, artist in enumerate(items['Items']):
-
-                            percent = int((float(start_index + index) / float(items['TotalRecordCount'])) * 100)
-                            message = artist['Name']
-                            dialog.update(percent, heading="%s: %s" % (translate('addon_name'), library['Name']), message=message)
-                            obj.artist(artist, library=library)
-
-                            for albums in server.get_albums_by_artist(artist['Id']):
-
-                                for album in albums['Items']:
-                                    obj.album(album)
-
-                                    for songs in server.get_items(album['Id'], "Audio"):
-                                        for song in songs['Items']:
-
-                                            dialog.update(percent,
-                                                          message="%s/%s/%s" % (message, album['Name'][:7], song['Name'][:7]))
-                                            obj.song(song)
-
-                            for songs in server.get_songs_by_artist(artist['Id']):
-                                for song in songs['Items']:
-
-                                    dialog.update(percent, message="%s/%s" % (message, song['Name']))
-                                    obj.song(song)
-
+                    obj = Music(self.server, jellyfindb, musicdb, self.direct_path, library)
+
+                    library_id = library['Id']
+
+                    # Get all items in the library to process locally
+                    artists_data = server.get_artists(library_id)
+                    artists = artists_data['Items']
+                    num_artists = artists_data['TotalRecordCount']
+                    albums = server.get_library_items(library_id, 'MusicAlbum')['Items']
+                    songs = server.get_library_items(library_id, 'Audio')['Items']
+
+                    for index, artist in enumerate(artists):
+                        artist_name = artist.get('Name')
+
+                        # Update percentage dialog
+                        percent = int((float(index) / float(num_artists)) * 100)
+                        message = artist_name
+                        dialog.update(percent, heading="%s: %s" % (translate('addon_name'), library['Name']), message=message)
```

Comment:
Why we directly assigning one variable name to another. Couldn't you just put the `artist_name` in the message value?

Reply:
Mostly blind copy/paste from the previous code for this part